### DISCUSSION TITLE
issue: defaultValues or reset not accepting 32 or 36 character hex string (dashes excluded/included) or a 16 byte binary Buffer in React Native App.

### DISCUSSION THREAD

**Discussion by saravanakumargn** [react-hook-form/react-hook-form#8456] [Q&A] [answered]
### Version Number

7.31.3

### Codesandbox/Expo snack

https://github.com/saravanakumargn/HookFormBsonProject

### Steps to reproduce

1. Goto https://github.com/saravanakumargn/HookFormBsonProject/blob/main/App.js
2. Uncomment #L28 https://github.com/saravanakumargn/HookFormBsonProject/blob/main/App.js#L28
3. Run the App on ios/Android
4. See an error in CLI `TypeError: undefined is not an object (evaluating 'buffer.toString')`

### Expected behaviour

1. In #L19 https://github.com/saravanakumargn/HookFormBsonProject/blob/main/App.js#L19 directly `new UUID()` is assigned to `_id`. (Which is not working)
2. In #L20 https://github.com/saravanakumargn/HookFormBsonProject/blob/main/App.js#L20 UUID value converted into a string and assigned to '_idString`. (Which is working fine);

Current Output CLI: 
![Screenshot 2022-06-05 at 7 59 30 PM](https://user-images.githubusercontent.com/1115851/172055567-958e9e4d-d1e8-43cb-8b6b-ee9881d36818.png)

Expected Output:
`defaultValues` or `reset` should accept UUID or any string buffer value and the value should be usabele when I access using `getValues('_id')`

Node JS Version: v16.15.1

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
can you reproduce it in here: https://snack.expo.io/@bluebill1049/react-hook-form-v7---controller?

**Reply by saravanakumargn** [reply]
Not able to reproduce it in expo snack. I am facing this only in React Native. Please check the code in https://github.com/saravanakumargn/HookFormBsonProject
